cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -406,6 +406,31 @@ public void executeFlashingChange() throws RetrofitError {
         });
     }
 
+    @WorkerThread
+    public void flashCodeFile(final File file) throws ParticleCloudException {
+        performFlashingChange(new FlashingChange() {
+            @Override
+            public void executeFlashingChange() throws RetrofitError {
+                mainApi.flashFile(deviceState.deviceId,
+                        new TypedFile("multipart/form-data", file));
+            }
+        });
+    }
+
+
+    @WorkerThread
```

Comment:
Much as appreciate streaming APIs, what's the intent here?  Was there a common use case you're trying to address here where the `File` method?

Also, why does this `flashCodeFile()` still use `application/octet-stream` as the mime type, while the one that takes a `File` arg specifies `multipart/form-data`?  Is that intentional?